datahub-graphql-core/src/main/resources/gms.graphql

```diff
@@ -1318,4 +1327,405 @@ enum ChartQueryType {
     LookML
     """
     LOOKML
-}
\ No newline at end of file
+}
+
+type MLModel implements Entity {
+
+    """
+    The unique mlmodel URN
+    """
+    urn: String!
+
+    """
+    GMS Entity Type
+    """
+    type: EntityType!
+
+    """
+    MLModel native name
+    """
+    name: String!
+
+    """
+    Fabric type where mlmodel belongs to or where it was generated
+    """
+    origin: FabricType!
+
+    """
+    Human readable description for mlmodel
+    """
+    description: String
+
+    """
+    Tags used for searching mlmodel
+    """
+    tags: [String!]!
+
+    """
+    Ownership metadata of the mlmodel
+    """
+    ownership: Ownership
+
+    """
+    ModelProperties metadata of the mlmodel
+    """
+    mlModelProperties: MLModelProperties
+
+    """
+    Intended use of the mlmodel
+    """
+    intendedUse: IntendedUse
+
+    """
+    Factors metadata of the mlmodel
+    """
+    mlModelFactorPrompts: MLModelFactorPrompts
+
+    """
+    Metrics metadata of the mlmodel
+    """
+    metrics: Metrics
+
+    """
+    Evaluation Data of the mlmodel
+    """
+    evaluationData: [BaseData]
+
+    """
+    Training Data of the mlmodel
+    """
+    trainingData: [BaseData]
+
+    """
+    Quantitative Analyses of the mlmodel
+    """
+    quantitativeAnalyses: QuantitativeAnalyses
+
+    """
+    Ethical Considerations of the mlmodel
+    """
+    ethicalConsiderations: EthicalConsiderations
+
+    """
+    Caveats and Recommendations of the mlmodel
+    """
+    caveatsAndRecommendations: CaveatsAndRecommendations
+
+    """
+    References to internal resources related to the mlmodel
+    """
+    institutionalMemory: InstitutionalMemory
+
+    """
+    Source Code
+    """
+    sourceCode: SourceCode
+
+    """
+    Status metadata of the mlmodel
+    """
+    status: Status
+
+    """
+    Cost Aspect of the mlmodel
+    """
+    cost: Cost
+
+    """
+    Deprecation
+    """
+    deprecation: Deprecation
+}
+
+type MLModelProperties {
+
+    description: String
+
+    date: Long
+
+    version: String
```

Comment:
All of these are optional??

Reply:
Yes - https://github.com/linkedin/datahub/blob/master/gms/api/src/main/pegasus/com/linkedin/ml/MLModel.pdl

Other than urn and those fields part of urn all other aspects are Optional.